runtime/src/main/java/com/dylibso/chicory/runtime/Memory.java

```diff
@@ -1,363 +1,106 @@
-    }
-
-    public void writeF32(int addr, float data) {
-        try {
-            buffer.putFloat(addr, data);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public long readF32(int addr) {
-        try {
-            return buffer.getInt(addr);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public float readFloat(int addr) {
-        try {
-            return buffer.getFloat(addr);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public void writeF64(int addr, double data) {
-        try {
-            buffer.putDouble(addr, data);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public double readDouble(int addr) {
-        try {
-            return buffer.getDouble(addr);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public long readF64(int addr) {
-        try {
-            return buffer.getLong(addr);
-        } catch (IndexOutOfBoundsException e) {
-            throw new WasmRuntimeException("out of bounds memory access");
-        }
-    }
-
-    public void zero() {
-        this.fill((byte) 0);
-    }
-
-    public void fill(byte value) {
-        fill(value, 0, buffer.capacity());
-    }
+    void writeString(int offset, String data, Charset charSet);
+
+    void writeString(int offset, String data);
+
+    String readString(int addr, int len);
+
+    String readString(int addr, int len, Charset charSet);
+
+    void writeCString(int offset, String str);
+
+    void writeCString(int offset, String str, Charset charSet);
+
+    String readCString(int addr, Charset charSet);
+
+    String readCString(int addr);
+
+    void write(int addr, byte[] data);
+
+    void write(int addr, byte[] data, int offset, int size);
+
+    byte read(int addr);
+
+    byte[] readBytes(int addr, int len);
+
+    void writeI32(int addr, int data);
+
+    int readInt(int addr);
+
+    long readI32(int addr);
+
+    long readU32(int addr);
+
+    void writeLong(int addr, long data);
+
+    long readLong(int addr);
+
+    long readI64(int addr);
+
+    void writeShort(int addr, short data);
+
+    short readShort(int addr);
+
+    long readI16(int addr);
+
+    long readU16(int addr);
+
+    void writeByte(int addr, byte data);
+
+    long readU8(int addr);
+
+    long readI8(int addr);
+
+    void writeF32(int addr, float data);
+
+    long readF32(int addr);
+
+    float readFloat(int addr);
+
+    void writeF64(int addr, double data);
+
+    double readDouble(int addr);
+
+    long readF64(int addr);
+
+    void zero();
+
+    void fill(byte value);
```

Comment:
I think these can be removed. The single call to `zero()` in `Instance.initialize()` doesn't seem needed. We can make the contract that `Memory` is filled with zero when it is created (which is what any sane implementation would do anyway). Then we don't need to redundantly fill it with zero again after creating it.

Reply:
removed :+1: